src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -625,48 +626,55 @@ private static string Encode5987(StringSegment input)
             ? stackalloc byte[MaxStackAllocSizeBytes]
             : bufferFromPool = ArrayPool<byte>.Shared.Rent(maxInputBytes);
 
+        var maxInputAsciiCount = Encoding.ASCII.GetMaxCharCount(input.Length);
```

Comment:
I'm missing something - isn't `input.Length` already in characters (vs bytes)?

Reply:
Yeah the UTF8 conversion confused me, but I don't think we need that line at all as you point out. Actually does more harm then good.